pkg/manager/internal.go

```diff
@@ -150,6 +154,45 @@ type controllerManager struct {
 	retryPeriod time.Duration
 }
 
+type errSignaler struct {
+	// errSignal indicates that an error occurred, when closed.  It shouldn't
+	// be written to.
+	errSignal chan struct{}
+
+	// err is the received error
+	err error
```

Comment:
Does it make sense to change it to `[]error`?
In `SignalError()`, we append errors.
In `Error()`, we aggregate errors into one error.

Reply:
hmm...since mgr bails out on seeing first error, converting to slice of errors is not going to be of much help.